src/Npgsql/Internal/TypeMapping/UserEnumTypeMappings.cs

```diff
@@ -21,16 +21,19 @@ sealed class UserEnumTypeMapping<TEnum> : IUserEnumTypeMapping
     public Type ClrType => typeof(TEnum);
     public INpgsqlNameTranslator NameTranslator { get; }
 
-    readonly Dictionary<TEnum, string> _enumToLabel = new();
-    readonly Dictionary<string, TEnum> _labelToEnum = new();
+    readonly Dictionary<TEnum, string> _enumToLabel;
+    readonly Dictionary<string, TEnum> _labelToEnum;
 
     public UserEnumTypeMapping(string pgTypeName, INpgsqlNameTranslator nameTranslator)
     {
         (PgTypeName, NameTranslator) = (pgTypeName, nameTranslator);
-
-        foreach (var field in typeof(TEnum).GetFields(BindingFlags.Static | BindingFlags.Public))
+        var fields = typeof(TEnum).GetFields(BindingFlags.Static | BindingFlags.Public);
+        _enumToLabel = new(fields.Length, new EnumEqualityComparer());
```

Comment:
Can you provide some background on this change, and specifically benchmarks to show that it improves speed?

BTW if we really want to optimize this, on .NET 7.0 we can probably use the new [frozen collections](https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8#performance-focused-types).

Reply:
@roji 
I did a research frozen dictionary didn't work, my solution didn't work either

Reply:
Are you saying that the approach introduced in this PR didn't lead to a perf improvement? And also that using the new frozen dictionary didn't?

Reply:
I'm not sure I understand - are you saying that using your technique - or frozen dictionary - led to a constant allocation reduction **when looking up** elements in the dictionary (as opposed to when constructing it, which is a one-time thing we don't care about)?